File: history.py
from typing import List
import move


class History(move.Move):
    def __init__(self, board: List[List[str]], mv: move.Move) -> None:
        super().__init__(mv.src_row, mv.src_column, mv.dest_row, mv.dest_column)
        self.data |= (ord(board[mv.src_row][mv.src_column]) << 12 |
                      ord(board[mv.dest_row][mv.dest_column]) << 19)

    @property
    def src_piece(self) -> str:
        return chr((self.data & 0x7f000) >> 12)

    @property
    def dest_piece(self) -> str:
        return chr((self.data & 0x3f80000) >> 19)

File: history.c
#include "history.h"



void    History_init(History* self, int move, char src_piece, char dest_piece, int white_score, int black_score) {
    self->move = move;
    self->src_piece = src_piece;
    self->dest_piece = dest_piece;
    self->white_score = white_score;
    self->black_score = black_score;
}
